src/shared/Microsoft.Git.CredentialManager/CredentialCacheStore.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System.Collections.Generic;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public class CredentialCacheStore : ICredentialStore
+    {
+        readonly IGit _git;
+        readonly IHelperProcess _helper;
+
+        public CredentialCacheStore(IGit git, IHelperProcess helper)
+        {
+            _git = git;
+            _helper = helper;
+        }
+
+        #region ICredentialStore
+
+        public ICredential Get(string service, string account)
+        {
+            var input = MakeGitCredentialsEntry(service, account);
+
+            var result = _helper.InvokeAsync(
+                _git.GitPath,
+                "credential-cache get",
+                input
+            ).GetAwaiter().GetResult();
+
+            if (result.ContainsKey("username") && result.ContainsKey("password"))
+            {
+                return new GitCredential(result["username"], result["password"]);
+            }
+
+            return null;
```

Comment:
I wonder if throwing an error, or a warning would be more appropriate if there's no user/pass in the result?

Also makes me wonder what exactly `git-credential-cache` _does_ when it doesn't find a cached credential?

Reply:
> I wonder if throwing an error [...] would be more appropriate

It's a good question. I was drafting off what the interface currently instructs:
https://github.com/microsoft/Git-Credential-Manager-Core/blob/791173e9dcd55ea7ba84645648a7aa06e8f7d166/src/shared/Microsoft.Git.CredentialManager/ICredentialStore.cs#L16

The primary callsite seems to be in `HostProvider` where it branches on `null`:
https://github.com/microsoft/Git-Credential-Manager-Core/blob/8a699dbc9af32277b3ff721e65d5e3efb7919603/src/shared/Microsoft.Git.CredentialManager/HostProvider.cs#L131

I guess the functional people would tell us to use a `Maybe` here, but since that's not really the style of this codebase, I think leaving it returning `null` is probably fine.

`credential-cache` is a credential helper, so it [should produce nothing](https://git-scm.com/docs/gitcredentials):
> A helper is free to produce a subset, or even no values at all if it has nothing useful to provide.